xtask/src/main.rs

```diff
@@ -32,145 +32,63 @@ use crate::{
     codegen::Mode,
     dist::DistCmd,
     install::{InstallCmd, Malloc, ServerOpt},
-    metrics::MetricsCmd,
-    pre_cache::PreCacheCmd,
-    release::{PromoteCmd, ReleaseCmd},
 };
 
 fn main() -> Result<()> {
     let _d = pushd(project_root())?;
 
-    let mut args = Arguments::from_env();
-    let subcommand = args.subcommand()?.unwrap_or_default();
-
-    match subcommand.as_str() {
-        "install" => {
-            if args.contains(["-h", "--help"]) {
-                eprintln!(
-                    "\
-cargo xtask install
-Install rust-analyzer server or editor plugin.
-
-USAGE:
-    cargo xtask install [FLAGS]
-
-FLAGS:
-        --client[=CLIENT] Install only VS Code plugin.
-                          CLIENT is one of 'code', 'code-exploration', 'code-insiders', 'codium', or 'code-oss'
-        --server          Install only the language server
-        --mimalloc        Use mimalloc allocator for server
-        --jemalloc        Use jemalloc allocator for server
-    -h, --help            Prints help information
-        "
-                );
-                return Ok(());
-            }
-            let server = args.contains("--server");
-            let client_code = args.contains("--client");
-            if server && client_code {
+    let flags = flags::Xtask::from_env()?;
+    match flags.subcommand {
+        flags::XtaskCmd::Help(_) => {
+            println!("{}", flags::Xtask::HELP);
+            return Ok(());
+        }
+        flags::XtaskCmd::Install(flags) => {
+            if flags.server && flags.client {
                 eprintln!(
                     "error: The argument `--server` cannot be used with `--client`\n\n\
                      For more information try --help"
                 );
                 return Ok(());
             }
 
-            let malloc = if args.contains("--mimalloc") {
+            let malloc = if flags.mimalloc {
                 Malloc::Mimalloc
-            } else if args.contains("--jemalloc") {
+            } else if flags.jemalloc {
                 Malloc::Jemalloc
             } else {
                 Malloc::System
             };
 
-            let client_opt = args.opt_value_from_str("--client")?;
-
-            finish_args(args)?;
+            let client_bin = flags.code_bin.map(|it| it.parse()).transpose()?;
 
             InstallCmd {
-                client: if server { None } else { Some(client_opt.unwrap_or_default()) },
-                server: if client_code { None } else { Some(ServerOpt { malloc }) },
+                client: if flags.server { None } else { client_bin },
```

Comment:
With this change, if you do not specify `code_bin` explicitly, the client will not be installed. Was that on purpose?